### ISSUE TITLE
About transform_models_if_DDP

### ISSUE THREAD

**Issue by Hzj199** [bmaltais/kohya_ss#1703]
I have two questions:
1. Why do I need to use `transform_models_if_DDP `after accelerator.prepare?  
2. `avg_loss = accelerator.gather(loss.repeat(args.train_batch_size)).mean()` (in diffusers) is not used before `accelerator.backward(loss)`
I found that there seemed to be no communication between multiple GPUs, I tried to save the models in different processes, and the weights of the models were not consistent.
